tripple_range angle
tripple_range speed

